kernel/include/kernel_internal.h

```diff
@@ -143,7 +143,7 @@ extern void smp_timer_init(void);
 
 extern void z_early_boot_rand_get(uint8_t *buf, size_t length);
 
-#if CONFIG_STACK_POINTER_RANDOM
+#if defined(CONFIG_STACK_POINTER_RANDOM) && CONFIG_STACK_POINTER_RANDOM != 0
```

Comment:
For consistency, I recommend adding ()'s around the "CONFIG_STACK_POINTER_RANDOM != 0" check. This applies to each of the other instances of that check in this commit as well (4 in total).